File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Order extends Model
{
    use HasFactory;

    protected $fillable = [
        'client_id',
        'code',
        'table_id',
        'waiter_id',
        'reservation_id',
        'price',
        'status',
        'special_request',
        'instructions',
        'address',
        'phone',
        'order_type',
        'start_time',
        'end_time'
    ];

    public function users()
    {
        return $this->belongsToMany(User::class);
    }

    public function reservation()
    {
        return $this->belongsToMany(Reservation::class);
    }

    public function orderItems()
    {
        return $this->belongsToMany(MenuItem::class, 'menu_items_orders')
            ->withPivot('id','order_id','quantity', 'price','menu_item_id')
            ->withTimestamps();
    }

}

File: mobile/app/src/main/java/com/yehorsk/platea/orders/domain/models/Order.kt
package com.yehorsk.platea.orders.domain.models

import com.yehorsk.platea.core.utils.UiText

data class Order(
    val id: String,
    val code: String,
    val createdAt: String,
    val updatedAt: String,
    val clientId: Int? = null,
    val tableId: Int? = null,
    val waiterId: Int? = null,
    val reservationId: Int? = null,
    val price: Double,
    val status: String,
    val phone: String? = null,
    val specialRequest: String? = null,
    val address: String? = null,
    val instructions: String? = null,
    val orderType: Int,
    val startTime: String,
    val endTime: String,
    val date: String? = null,
    val completedAt: String? = null,
    val orderItems: List<OrderMenuItem> = emptyList<OrderMenuItem>()
)

data class SectionedOrders(
    val date: UiText,
    val orders: List<Order>
)